Feature rsv001 reservation form

andress20 opened the pull request:
**Create Reservation Form.**

- [x] Create form (branch, date, time, time-range, how many people and agree restaurant times)
- [x] install font awesome icons
- [x] link Restaurant list with Reservation Form (mock data)
- [x] create path /reservation

* links sanitary register don't work just icon
* submit button doesn't have functionality yet

![reservationPicture](https://user-images.githubusercontent.com/8940946/99456121-da08ae80-28f6-11eb-96bd-a8ba1cc96f8b.png)

andress20 committed:
create ReservationForm.js

andress20 committed:
Create Reservation Form, install font awesome icons

simonhoyos commented on alamesa/src/pages/ReservationForm.js:
```diff
@@ -0,0 +1,298 @@
+                  autoFocus
+                  required
+                > 
+                  <option 
```
Podemos tener un arreglo de opciones y luego renderizar la lista de opciones dinámicamente. En caso que queramos agregar un locación en el futuro solo es agregarla al arrreglo

simonhoyos commented on alamesa/src/pages/ReservationForm.js:
```diff
@@ -0,0 +1,298 @@
+                <Label
+                  className="Form__reservation-branch-label"
+                  htmlFor="branch"
+                >Sucursal
```
Pasaria Sucursal para la siguiente línea. Lo mismo para todo el archivo

andress20 committed:
fix:
branches in list
time-range in list
now option lists render dynamically

andress20 committed:
Merge branch 'develop' into feature-RSV001-ReservationForm

carosalazar28 commented on alamesa/src/App.js:
```diff
@@ -32,6 +31,7 @@ function App () {
         <Route exact path='/restaurant-profile' component={RestaurantForm}/>
         <Route path="/your-profile" component={RestaurantProfile}/>
         <Route exact path="/restaurants" component={Restaurants}/>
+        <Route exact path="/reservation" component={Reservation}/>
```
En la ruta se puede encadenar restaurants/reservation ?

simonhoyos replied:
podría ser incluso `/restaurants/:restaurantId/reservation`. Así van a saber a cuál restaurante se está haciendo la reserva

andress20 replied:
Ese id seria el que tenemos generado con uuid?
quedaria algo asi entonces:

path = /restaurants/**17f38045-5ebc-43c0-8608-f4adb663afa2**/reservation

es correcto?

andress20 committed:
inclue:
restaurantId in the url path
restaurantId pass via state to the next component in the submit buttom

andress20 committed:
Merge branch 'feature-RSV001-ReservationForm' of github.com:danielsespitia/proyecto-top into feature-RSV001-ReservationForm
merge necessary because conflicts between my remote repo waitin for approve and some changes that I make

andress20 committed:
RestaurantId in  URL

simonhoyos commented on alamesa/src/App.js:
```diff
@@ -32,6 +31,7 @@ function App () {
         <Route exact path='/restaurant-profile' component={RestaurantForm}/>
         <Route path="/your-profile" component={RestaurantProfile}/>
         <Route exact path="/restaurants" component={Restaurants}/>
+        <Route exact path="/restaurants/:Restaurant_id/reservation" component={Reservation}/>
```
tratar de usar camelCase para lo que sea JS, `restaurantId`

simonhoyos commented on alamesa/src/components/Restaurant.js:
```diff
@@ -38,11 +40,18 @@ function Restaurant ( { name, logo } ) {
+        <Link to={{
+          pathname: `/restaurants/${id}/reservation`,
+          state: {
+            Restaurant: { id, name, logo},
```
Ya este estado no es necesario, el id del restaurante ya va en la url. Idealmente, si necesitamos más información del resturante, la deberiamos pedir al BE

andress20 replied:
Simon. Como aun no tenemos la implementacion de traer info del BE, esto lo podemos configurar luego? y por ahora dejar el `name` y `logo` ya que si los estamos renderizando con la data mockeada.

simonhoyos commented on alamesa/src/pages/ReservationForm.js:
```diff
@@ -0,0 +1,333 @@
+      agree,
+    } = this.state
+
+    const { id, name, logo } = this.props.location.state.Restaurant
```
La info del restaurante la deberiamos pedir desde el servidor. En este caso para que la información se muestre siempre tengo que navegar desde Restaurant a esta ruta. Qué pasa si yo pongo la URL manualmente? la info no me llega y no se muestra ni el nombre, ni el logo

andress20 replied:
Por el momento quedo apuntando al primer elemento de la data mockeada siempre. Despues con la implementacion de la BD modificacmos ese comportamiento.

andress20 committed:
Solving url path params

andress20 committed:
Restaurant reservation render first restaurant always
this behavior is for now while implement db later

andress20 committed:
Merge branch 'develop' into feature-RSV001-ReservationForm
